davydsonsantana/social-media
# feat/add edit post endpoint

## Description



Implementation of update flow of post entity.

## Changes Made



- Added Post PATCH endpoint
- Configured the Mongo polymorphism to parse BaseEvent and its derivations.

## Testing



- Manual testing.
- Trigger creation of new post endpoint
- Identification of created post ID
- Trigger creation of edit post endpoint
- Check eventstore events
- Check table post on read model 

## Checklist


## Screenshots (if applicable)



### Post creation
![image](https://github.com/davydsonsantana/social-media/assets/1733168/3d9fc28d-5f6a-4f27-a263-5d7bb432e0a1)
![image](https://github.com/davydsonsantana/social-media/assets/1733168/c5426b9a-edb5-422c-bc22-1da505e0521d)
![image](https://github.com/davydsonsantana/social-media/assets/1733168/2b64a35f-6413-4446-baf2-0cbf7bd7a26f)

### Post editing
![image](https://github.com/davydsonsantana/social-media/assets/1733168/e384e400-bd87-4705-aed6-4cc981d51ee6)
![image](https://github.com/davydsonsantana/social-media/assets/1733168/2d6efe1d-8f00-4ac4-96e6-2b52c2e1b2cc)
![image](https://github.com/davydsonsantana/social-media/assets/1733168/7fad92c7-2607-4d7d-aa05-4cafc8e0018d)
